test/test_jit.py

```diff
@@ -15187,6 +15187,25 @@ def test_dict_types(self):
             def foo():
                 new_item = {'score': [1.0], 'ys': [1, 2, 3]}
 
+    def test_dict_invalid_annotations(self):
+        # Check for invalid value type annotation
+        with self.assertRaisesRegex(AssertionError, "Unsupported annotation"):
+            @torch.jit.script
```

Comment:
Nit:
Define the function outside of `self.assertRaisesRegex` and only script it inside in order to minimize amount of code under assertion.

Reply:
Looks like most our tests use it the way I wrote it. We should plan on adding them as part of our Better engineering effort :)